provisioning/templates-api/README.md

```diff
@@ -0,0 +1,58 @@
+# Provisioning of the Templates API
+
+## Directory Structure
+
+Entrypoint:
+- `deploy.sh` - entrypoint for the AWS and Azure deployments.
+- `deploy_local.sh` - entrypoint for the local deployment.
+
+
+Directories:
+- `dev` - various auxiliary files for development.
+- `docker` - docker image of the service.
+- `etcd` - configuration of the etcd cluster.
+- `kubernets` - templates of the Kubernetes configurations.
+
+Included files (they are not called directly):
+- `common.sh` - main file for all deployments.
+- `aws.sh` - deploy script for AWS stacks.
+- `azure.sh` - deploy script for Azure stacks.
+- `local.sh` - deploy script for a local MiniKube deployment.
+
+## Production Deployment
+
+- A tag `templates-api-vX.Y.Z` triggers GitHub Workflow.
+- The `build-and-push-templates-api` step builds the API image and pushes it to a repository in AWS and Azure.
+- Push to Azure ACR triggers a release pipeline.
+- Links to the AWS and Azure pipelines can be found in the `Service` page, in the internal Confluence.
+
+## Local Deployment
+
+### Docker
+
+In most cases, it is enough to run the service locally via Docker.
+```sh
+docker-compose run --rm -u "$UID:$GID" --service-ports dev bas
+make run-templates-api
+```
+
+Read more in `docs/DEVELOPMENT.md`.
+
+### MiniKube
+
+If you need to debug or test something in a Kubernetes cluster, you can use local deployment using [MiniKube](https://minikube.sigs.k8s.io/docs/start/).
+```sh
+./provisioning/templates-api/deploy_local.sh
```

Comment:
Skuste prosim, ci aj vam to pojde bez problemov.
Staci mat MiniKube.

Reply:
BTW: prvy deploy trva dlhsie, kym sa zbuilduje image a rozbehne MiniKube, potom to uz idem celkom rychlo.

Reply:
Hmm, já mám lokální Kubernetes přes Docker Desktop a vím že dřív se to nesnášelo. Budu muset nějdřív pogooglit jak na to.

Reply:
ok, `minikube start --vm=true` to spustit vo VM, to by mohlo ist nezavisle.

Reply:
jo, byl tam nějaký clash s networkingem ale napodruhé to prošlo 👍